src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1086,11 +1075,9 @@ CHIP_ERROR CASESession::SendSigma2()
                                                 static_cast<uint32_t>(mEphemeralKey->Pubkey().Length())));
     ReturnErrorOnFailure(tlvWriterMsg2.PutBytes(TLV::ContextTag(4), msg_R2_Encrypted.Get(),
                                                 static_cast<uint32_t>(msg_r2_signed_enc_len + CHIP_CRYPTO_AEAD_MIC_LENGTH_BYTES)));
-    if (mLocalMRPConfig.HasValue())
-    {
-        ChipLogDetail(SecureChannel, "Including MRP parameters");
-        ReturnErrorOnFailure(EncodeMRPParameters(TLV::ContextTag(5), mLocalMRPConfig.Value(), tlvWriterMsg2));
-    }
+
+    ReturnErrorOnFailure(EncodeSessionParameters(TLV::ContextTag(5), mLocalMRPConfig, tlvWriterMsg2));
```

Comment:
would it make sense to add these ContextTags to some sigma tags header and share around?
I also realinze this is not your PR directly, so followup (or just issue) is fine as well.

Reply:
https://github.com/project-chip/connectedhomeip/issues/30476

Filling issue. I don't want to add to this PR as it is already very large and still needs another approval. Want to keep this PR focused on Session Parameters only